src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/KustoSinkTask.java

```diff
@@ -162,8 +141,8 @@ public static Map<String, TopicIngestionProperties> getTopicsToIngestionProps(Ku
                 }
                 TopicIngestionProperties topicIngestionProperties = new TopicIngestionProperties();
                 topicIngestionProperties.ingestionProperties = props;
-                topicIngestionProperties.streaming = streaming;
-                result.put(mapping.getString("topic"), topicIngestionProperties);
+                topicIngestionProperties.streaming = mapping.isStreaming();
+                result.put(mapping.getTopic(), topicIngestionProperties);

```

Comment:
org.json.getString used to throw JSONException when null or empty. Those empty/null checks needs to be explicitly handled in jackson I believe. Since mapping.getTopic() is a mandatory, Can we have a notnull /notempty check for mapping.getTopic() and a configexception when empty.

Reply:
Thanks for spotting this. I added the validation in the config, after derializing the mappings